dpnp/backend/extensions/lapack/evd_batch_common.hpp

```diff
@@ -0,0 +1,138 @@
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//*****************************************************************************
+
+#pragma once
+
+#include <oneapi/mkl.hpp>
+#include <pybind11/pybind11.h>
+
+// dpctl tensor headers
+#include "utils/type_dispatch.hpp"
+
+#include "evd_common_utils.hpp"
+#include "types_matrix.hpp"
+
+namespace dpnp::extensions::lapack::evd
+{
+typedef sycl::event (*evd_batch_impl_fn_ptr_t)(
+    sycl::queue &,
+    const oneapi::mkl::job,
+    const oneapi::mkl::uplo,
+    const std::int64_t,
+    const std::int64_t,
+    char *,
+    char *,
+    const std::vector<sycl::event> &);
+
+namespace dpctl_td_ns = dpctl::tensor::type_dispatch;
+namespace py = pybind11;
+
+template <typename dispatchT>
+std::pair<sycl::event, sycl::event>
+    evd_batch_func(sycl::queue &exec_q,
+                   const std::int8_t jobz,
+                   const std::int8_t upper_lower,
+                   dpctl::tensor::usm_ndarray &eig_vecs,
+                   dpctl::tensor::usm_ndarray &eig_vals,
+                   const std::vector<sycl::event> &depends,
+                   const dispatchT &evd_batch_dispatch_table)
+{
+    const int eig_vecs_nd = eig_vecs.get_ndim();
+
+    const py::ssize_t *eig_vecs_shape = eig_vecs.get_shape_raw();
+    const py::ssize_t *eig_vals_shape = eig_vals.get_shape_raw();
+
+    const int expected_eig_vecs_nd = 3;
+    const int expected_eig_vals_nd = 2;
+
+    common_evd_checks(exec_q, eig_vecs, eig_vals, eig_vecs_shape,
+                      expected_eig_vecs_nd, expected_eig_vals_nd);
+
+    if (eig_vecs_shape[2] != eig_vals_shape[0] ||
+        eig_vecs_shape[0] != eig_vals_shape[1])
+    {
+        throw py::value_error(
+            "The shape of 'eig_vals' must be (batch_size, n), "
+            "where batch_size = " +
+            std::to_string(eig_vecs_shape[0]) +
+            " and n = " + std::to_string(eig_vecs_shape[1]));
+    }
+
+    size_t src_nelems(1);
+
+    for (int i = 0; i < eig_vecs_nd; ++i) {
+        src_nelems *= static_cast<size_t>(eig_vecs_shape[i]);
+    }
+
+    if (src_nelems == 0) {
+        // nothing to do
+        return std::make_pair(sycl::event(), sycl::event());
+    }
```

Comment:
That part is also common, can we move it to separate function, like:
```cpp
    if (get_src_nelems(eig_vecs_nd, eig_vecs_shape) == 0) {
        // nothing to do
        return std::make_pair(sycl::event(), sycl::event());
    }
```

Reply:
I have added `check_zeros_shape` function.
Thanks